jetbrains-core/src/software/aws/toolkits/jetbrains/core/explorer/devToolsTab/nodes/actions/SonoLogin.kt

```diff
@@ -6,18 +6,31 @@ package software.aws.toolkits.jetbrains.core.explorer.devToolsTab.nodes.actions
 import com.intellij.icons.AllIcons
 import com.intellij.openapi.actionSystem.AnActionEvent
 import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.application.runInEdt
 import com.intellij.openapi.project.DumbAwareAction
-import software.aws.toolkits.jetbrains.core.credentials.sono.SonoCredentialManager
+import software.aws.toolkits.jetbrains.core.credentials.ToolkitConnectionManager
+import software.aws.toolkits.jetbrains.core.credentials.pinning.CodeCatalystConnection
 import software.aws.toolkits.jetbrains.core.explorer.refreshDevToolTree
+import software.aws.toolkits.jetbrains.core.gettingstarted.requestCredentialsForCodeCatalyst
 import software.aws.toolkits.telemetry.UiTelemetry
 
 class SonoLogin : DumbAwareAction(AllIcons.Actions.Execute) {
     override fun actionPerformed(e: AnActionEvent) {
         val project = e.project ?: return
         UiTelemetry.click(project, elementId = "auth_start_CodeCatalyst")
+
         ApplicationManager.getApplication().executeOnPooledThread {
-            SonoCredentialManager.loginSono(project)
-            project.refreshDevToolTree()
+            val connectionManager = ToolkitConnectionManager.getInstance(project)
+            connectionManager.activeConnectionForFeature(CodeCatalystConnection.getInstance())?.let {
+                project.refreshDevToolTree()
```

Comment:
we have a connection but it may not be valid
```suggestion
                reauthConnectionIfNeeded(project, it)
                project.refreshDevToolTree()
```